src/ex.rs

```diff
@@ -417,13 +417,51 @@ where
     r
 }
 
+/// Checks if `-Z offline` is available if it is it updates the registry and returns true.
+/// If it is not available this returns false.
+fn prep_offline_registry(ex: &Experiment, toolchain: &Toolchain) -> Result<bool> {
+    let args = &[
+        "generate-lockfile",
+        "--manifest-path",
+        "Cargo.toml",
+        "-Z",
+        "offline",
+    ];
+    // This nop cargo command is to update the registry
+    // so we don't have to do it for each crate.
+    let args = &["cargo", "search", "lazy_static"];
+    let _ = toolchain.run_cargo(&ex.name, Path::new("."), args, CargoState::Unlocked);
+    let c = ex.crates()?
+        .iter()
+        .find(|cr| {
+            if let &ExCrate::Version {
+                ref name,
+                ref version,
+            } = cr
+            {
+                name == "lazy_static"
+            } else {
+                false
+            }
+        })
+        .unwrap();
```

Comment:
Could you use `expect` here? I believe we compile with debug info enabled, but either way, that'll be cleaner than having to hunt this down later...

Reply:
Will do, just sparring with ci. I wish I could build locally. Thank you for the ci.